src/axom/primal/geometry/BoundingBox.hpp

```diff
@@ -436,18 +436,19 @@ template <typename OtherType>
 AXOM_HOST_DEVICE bool BoundingBox<T, NDIMS>::intersectsWith(
   const BoundingBox<OtherType, NDIMS>& otherBB) const
 {
-  bool status = true;
-
   // AABBs cannot intersect if they are separated along any dimension
   for(int i = 0; i < NDIMS; ++i)
   {
-    status &= detail::intersect_bbox_bbox(m_min[i],
-                                          m_max[i],
-                                          otherBB.m_min[i],
-                                          otherBB.m_max[i]);
-  }  // END for all dimensions
+    if(!detail::intersect_bbox_bbox(m_min[i],
```

Comment:
👍 

Nice optimization

Reply:
Thanks! For the "naive" self-intersection algorithm in the tutorial, the `intersect(BoundingBox,BoundingBox)` test used something like half the cycles on a test that ran for a few seconds. This optimization improved the runtime by something like 30% on our blueos Power9s.

I still need to benchmark this before merging to ensure that it didn't give up performance on Intel runs.

Tag: @publixsubfan

Reply:
I kept the updated bbox intersect changes.
We can revisit if we need to use-platform specific overrides in a follow-up PR.

This is tracked in https://github.com/LLNL/axom/issues/582